api/lib/domain/validators/schooling-registration-validator.js

```diff
@@ -26,18 +31,25 @@ const validationSchema = Joi.object({
   }),
   birthCityCode: Joi.alternatives().conditional('birthCountryCode', {
     is: FRANCE_COUNTRY_CODE,
-    then: Joi.string().length(CITY_CODE_LENGTH).required(),
+    then: Joi.string()
+      .length(CITY_CODE_LENGTH)
+      .regex(INSEE_REGEX.FRANCE_CITY_CODE)
+      .required(),
     otherwise: Joi.string().length(CITY_CODE_LENGTH).optional(),
```

Comment:
```suggestion
    otherwise: Joi.string().optional(),
```
parce qu'en vrai, à l'étranger, on connaît pas le format des villes

Reply:
j'ai changé pour un `max(MAX_LENGTH)` au lieu du length. comme évoqué ce matin.